compiler/z/objectfmt/OMRJitCodeRWXObjectFormat.cpp

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include "codegen/CodeGenerator.hpp"
+#include "codegen/MemoryReference.hpp"
+#include "env/FilePointerDecl.hpp"
+#include "il/MethodSymbol.hpp"
+#include "il/SymbolReference.hpp"
+#include "infra/Assert.hpp"
+#include "objectfmt/FunctionCallData.hpp"
+#include "objectfmt/JitCodeRWXObjectFormat.hpp"
+#include "objectfmt/OMRJitCodeRWXObjectFormat.hpp"
+#include "runtime/CodeMetaDataManager.hpp"
+#include "runtime/Runtime.hpp"
+#include "z/codegen/CallSnippet.hpp"
+#include "z/codegen/S390GenerateInstructions.hpp"
+
+TR::Instruction *
+OMR::Z::JitCodeRWXObjectFormat::emitFunctionCall(TR::FunctionCallData &data)
+   {
+   TR::SymbolReference *callSymRef = data.methodSymRef;
+   TR::CodeGenerator *cg = data.cg;
+
+   if (callSymRef == NULL)
+      {
+      TR_ASSERT_FATAL(data.runtimeHelperIndex > 0, "FunctionCallData does not contain symbol reference for call or valid TR_RuntimeHelper");
```

Comment:
This should also be `TR_ASSERT_FATAL_WITH_NODE` similarly to how it is in the other constructor.